assets/src/util/sentry.tsx

```diff
@@ -19,6 +19,65 @@ const captureException = (ex: unknown, options?: RavenOptions) => {
   Raven.captureException(ex, options);
 };
 
+const offline_events_key = "sentry_offline_events";
+
+// Store offline Sentry events in localStorage for later retry
+function cacheEvent(options: RavenTransportOptions) {
+  try {
+    const cached = localStorage.getItem(offline_events_key);
+    const events = cached ? JSON.parse(cached) : [];
+    events.push({
+      ...options,
+      data: {
+        ...options.data,
+        extra: {
+          ...(options.data.extra || {}),
+          cached: new Date().toISOString(),
+        },
+      },
+    });
+    localStorage.setItem(offline_events_key, JSON.stringify(events));
```

Comment:
Should we have a limit on the number of events cached? The official offline transport does this by default and sets the limit at 30 events. If a screen was offline for a long time, it could accumulate a huge number of events, and trying to send them might bog down the connection or use up our event quota.